### DISCUSSION TITLE
Enabling GZip compression in quarqus http application.

### DISCUSSION THREAD

**Discussion by 16raghu** [quarkusio/quarkus#37474] [Q&A]
We have a quarkus application running on Quarkus 2.16.6.Final and we are using `io.quarkus:quarkus-resteasy-jsonb` for REST service.  We are trying to enable compression for the endpoints.
We had enabled the compression knobs[1] but we were still not able to compress the api response.  

When we were debugging we found that the response servlet had the `Content-Encoding` header set to `Identity`, which might have caused to skip compression encoding. 
On further debugging, we found that this code [here](https://github.com/quarkusio/quarkus/blob/3a328dc8562edb58540d16bf4896945941a4033d/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java#L397), might be probable reason.
Here it was mentioned that we can remove this header, what would be the best way to remove `Content-Encoding` header without any side effects? (Can we remove it in one of the web filter?) 

[1] HTTP Compression properties in `application.properties`.
```
......
quarkus.http.enable-compression=true
quarkus.http.compress-media-types=text/html,text/plain,text/xml,text/css,text/javascript,application/javascript,application/json
......
```

**Comment by mschorsch** [comment]
> The configuration property quarkus.http.enable-compression has no effect on compression support of RESTEasy Classic endpoints. 

https://quarkus.io/version/2.16/guides/resteasy#gzip-support

You can enable [GZip Support](https://quarkus.io/version/2.16/guides/resteasy#gzip-support) with the following properties:

```
quarkus.resteasy.gzip.enabled=true 
quarkus.resteasy.gzip.max-input=10M 
```

**Reply by 16raghu** [reply]
Thanks for looking into this. 

Those 2 properties is also set but api response were not encoded to GZIP. It would have no effect on compression as `Content-Encoding` header to `Identity`. 

If I forcefully remove the `Content-Encoding` header in one of the web filter then there is a compression to GZIP. 

I wanted to confirm if I can remove `Content-Encoding` header without any repercussions or what would be the best way to do this?

**Reply by mschorsch** [reply]
@geoand Sounds like a bug?

**Reply by geoand** [reply]
@mkouba would know more

**Reply by mkouba** [reply]
I beliebe that this was fixed in Quarkus 3.0+; see https://github.com/quarkusio/quarkus/pull/31438, https://github.com/quarkusio/quarkus/issues/26112 and https://github.com/quarkusio/quarkus/issues/31415.

I'm not really sure if we would be able to backport this fix in 2.16 though.

**Comment by 16raghu** [comment]
After moving to 3.6. Fixed gzip.
